pandas/io/formats/style.py

```diff
@@ -140,6 +150,11 @@ def __init__(self, data, precision=None, table_styles=None, uuid=None,
             precision = get_option('display.precision')
         self.precision = precision
         self.table_attributes = table_attributes
+        self.index_visible = index
+        if hidden_columns is None:
+            hidden_columns = []
+        self.hidden_columns = hidden_columns
```

Comment:
I actually wouldn't add these to the ``Styler.__init__`` prefer these as just methods. (like almost everything else). having more than 1 way to do things is not great.